src/routes/transactions/mappers/common/transaction-info.mapper.ts

```diff
@@ -103,35 +103,40 @@ export class MultisigTransactionInfoMapper {
       );
     }
 
-    if (this.isValidTokenTransfer(transaction)) {
-      const token = await this.tokenRepository.getToken(
+    if (!this.isValidTokenTransfer(transaction)) {
+      return this.customTransactionMapper.mapCustomTransaction(
+        transaction,
+        value,
+        dataSize,
         chainId,
-        transaction.to,
       );
+    }
 
-      if (token.type === TokenType.Erc20) {
+    const token = await this.tokenRepository
+      .getToken(chainId, transaction.to)
+      .catch(() => null);
+
+    switch (token?.type) {
+      case TokenType.Erc20:
         return this.erc20TransferMapper.mapErc20Transfer(
           token,
           chainId,
           transaction,
         );
-      }
-
-      if (token.type === TokenType.Erc721) {
+      case TokenType.Erc721:
         return this.erc721TransferMapper.mapErc721Transfer(
           token,
           chainId,
           transaction,
         );
-      }
+      default:
+        return this.customTransactionMapper.mapCustomTransaction(
+          transaction,
+          value,
+          dataSize,
+          chainId,
+        );
```

Comment:
Nitpick: IMO instead of inverting the `isValidTokenTransfer(...)` condition, and redirecting to `mapCustomTransaction` in two places, it could be simpler to write it this way (and the flow would be the same):

```javascript
if (this.isValidTokenTransfer(transaction)) {
  const token = await this.tokenRepository
    .getToken(chainId, transaction.to)
    .catch(() => null);

  switch (token?.type) {
    case TokenType.Erc20:
      return this.erc20TransferMapper.mapErc20Transfer(
        token,
        chainId,
        transaction,
      );
    case TokenType.Erc721:
      return this.erc721TransferMapper.mapErc721Transfer(
        token,
        chainId,
        transaction,
      );
  }
}

return this.customTransactionMapper.mapCustomTransaction(
  transaction,
  value,
  dataSize,
  chainId,
);
```

What do you think, do you find it more readable?

Reply:
I do! Pushed the suggestion [7decefc](https://github.com/safe-global/safe-client-gateway-nest/pull/375/commits/7decefc0b871d524f1bf79f177fa23aa56180fff)